modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -1378,14 +1387,39 @@ private void _processAttachment(
 				"File source " + fileSource + " is not supported");
 		}
 
-		com.liferay.portal.kernel.repository.model.FileEntry
-			serviceBuilderFileEntry = null;
-
 		byte[] fileContent = {};
 
 		if (fileEntry.getFileBase64() != null) {
 			fileContent = _decode(fileEntry.getFileBase64());
 		}
+		else if ((fileEntry.getFileSourceURL() != null) &&
+				 FeatureFlagManagerUtil.isEnabled("LPD-39967")) {
+
+			try {
+				URL url = new URL(fileEntry.getFileSourceURL());
+
+				if (Objects.equals(url.getProtocol(), "file")) {
+					throw new UnsupportedOperationException(
+						"Unsupported protocol");
+				}
+
+				URLConnection urlConnection = url.openConnection();
+
+				urlConnection.connect();
+
+				fileContent = StreamUtil.toByteArray(
```

Comment:
I think you should set the field to null once it has been downloaded as we do with the base64:

```java
fileEntry.setFileBase64(() -> (String)null);
```